File: result/translate/starchat/python2java/java/SUBARRAYS_DISTINCT_ELEMENTS.java
public static int f_gold ( int arr [ ] : int n ) {
        int s [ ] = new int [ 100000 ];
        int j = 0;
        int ans = 0;
        for ( int i = 0; i < n; i++ ) {
            while ( j < n && ( arr [ j ] not in s ) ) {
                s [ j ] = arr [ j ];
                j++;
            }
            ans += ( ( j - i ) * ( j - i + 1 ) ) / 2;
            s [ i ] = 0;
        }
        return ans;
    }

File: result/translate/starchat/java2python/python/SUBARRAYS_DISTINCT_ELEMENTS.py
def f_gold(arr, n):
        s = set()
        ans = 0
        j = 0
        for i in range(n):
            while j < n and arr[j] not in s:
                s.add(arr[j])
                j += 1
            ans += (j - i) * (j - i + 1) // 2
            s.remove(arr[i])
        return ans